### PR TITLE
Changes Static URL generation to utilize Django built-in Static template tag

### PR DESCRIPTION
**Pull request by ahackit**
This change does the following:
1. Removes the get_static_url from template tags
2. Adds Static template tag to locations previously using get_static_url 
3. Updates tests to account for this change

This allows for more robust static url generation to account for users who may be using cloud storage through Django static_storage

### CODE REVIEW

**Comment by jrief** [comment]
Oh, you must have found some legacy code from a very old version of Django :)

**Comment by ahackit** [comment]
Will rebase/revert the .gitignore and move this one on up once back from vacation. 

Appreciate the feedback!

**Comment by solarissmoke** [comment]
@ahackit any chance you have a few minutes to rebase this? I'll make sure it's merged promptly if you do.

**Comment by ahackit** [comment]
> @ahackit any chance you have a few minutes to rebase this? I'll make sure it's merged promptly if you do.

Apologize for the delay y'all. Give me 30 minutes and I'll have it complete.

File: .gitignore [hunk 1]
```diff
@@ -29,5 +29,4 @@ bin/
 include/
 lib/
 distribute*
-share/
-.DS_store
+share/
```
Comment (hartwork) [inline .gitignore]: Please revert that change, it does not seem related.

File: treebeard/tests/test_treebeard.py [hunk 2]
```diff
@@ -2355,24 +2355,14 @@ def test_treebeard_js(self):
...
-                    'src="/treebeard/treebeard-admin.js"></script>'
-                    '<script>(function($){'
+                    'src="{adminjs}"></script>'
+                    '<script>(function($){{'
                     'jQuery = $.noConflict(true);'
-                    '})(django.jQuery);</script>'
+                    '}})(django.jQuery);</script>'
                     '<script type="text/javascript" '
-                    'src="/treebeard/jquery-ui-1.8.5.custom.min.js"></script>')
+                    'src="{jqy}"></script>').format(adminjs=static('treebeard/treebeard-admin.js'), jqy=static('treebeard/jquery-ui-1.8.5.custom.min.js'))
```
Comment (hartwork) [inline treebeard/tests/test_treebeard.py:2363]: Do to curly braces in the JS code, I think `.format` is maybe not the best option to string interpolation here. How about

- a) `'..' % (..)`
- b) `".." + ... + ".."`

instead?

File: .gitignore [hunk 3]
```diff
@@ -29,5 +29,9 @@ bin/
 include/
 lib/
 distribute*
+<<<<<<< HEAD
 share/
 .DS_store
+=======
+share/
+>>>>>>> 6ce90a1a7bfb0383cddc5ca47efab0769f97a51d
```
Comment (hartwork) [inline .gitignore:37]: Please revert all changes to that file.

**Review by hartwork — CHANGES_REQUESTED** [review]
I have two comments below plus:
- Unless we're aiming for a squash-merge here, my vote for using `git rebase -i ...` to clean up the 9 commits to maybe three or so 3 without duplicate commit messages
- There is a merge commit in here. That would be nice to get rebased out
- Your commits use an e-mail that is not connected to your GitHub account. Is that on purpose?

File: .gitignore (hunk 3 above)
Comment (ahackit) [reply .gitignore]: Done. Apologize was dealing with some rebase goofyness I was working my way through.

Comment (hartwork) [reply .gitignore]: No worries, my bad. I was just about to publish my review when those came up. I probably should have waited a few moments more.

File: treebeard/tests/test_treebeard.py (hunk 2 above)
Comment (ahackit) [reply treebeard/tests/test_treebeard.py]: Updated along with commit clean ups

File: treebeard/tests/test_treebeard.py [hunk 4]
```diff
@@ -2356,12 +2355,12 @@ def test_treebeard_js(self):
         rendered = template.render(context)
         expected = ('<script type="text/javascript" src="jsi18n"></script>'
                     '<script type="text/javascript" '
-                    'src="/treebeard/treebeard-admin.js"></script>'
-                    '<script>(function($){'
+                    'src="' + static('treebeard/treebeard-admin.js') + '"></script>'
+                    '<script>(function($){{'
                     'jQuery = $.noConflict(true);'
-                    '})(django.jQuery);</script>'
+                    '}})(django.jQuery);</script>'
```
Comment (hartwork) [inline treebeard/tests/test_treebeard.py:2361]: These two lines have too many curly braces now, it seems.

**Review by hartwork — CHANGES_REQUESTED** [review]
A lot better already! :+1:

File: treebeard/tests/test_treebeard.py (hunk 4 above)
Comment (ahackit) [reply treebeard/tests/test_treebeard.py]: Removed!

**Comment by ahackit** [comment]
Apologize for the blood bath that was that rebase. Probably needed a second coffee. 

Tests should pass now, but will double check in a bit if they don't.

**Review by hartwork — APPROVED** [review]
(I'd massage the commit history more personally but) I think the overall diff is great now, thanks! :+1:

**Comment by solarissmoke** [comment]
@ahackit could you please fix the conflicts on master (these arose after #153 was merged). Alternatively if you check "allow edits from maintainers" I'm happy to fix up for you.

**Comment by ahackit** [comment]
@solarissmoke  Merge conflict has been resolved.

**Pull request merged**